src/Theme/utils.ts

```diff
@@ -0,0 +1,55 @@
+import { DefaultTheme } from 'styled-components';
+
+/**
+ * Helper function that returns a function that extracts values from the theme.
+ * Allows passing multiple strings as props to dig into nested values.
+ * This is intended for use in styled-comopnents's interpolation functions as
+ * an easier way to get properly typed values. Examples:
+ *
+ * instead of:
+ *
+ * styled.div`
+ *   margin-top: ${({theme}: {theme: DefaultTheme}):string => theme.ws.xsmall}
+ * `
+ *
+ * use instead:
+ *
+ * styled.div`
+ *   margin-top: ${fromTheme('ws', 'xsmall')}
+ */
+
+export const fromTheme = (...levels: string[]) => (
+  ({ theme }: { theme: DefaultTheme }): string => {
+    const propList = [...levels];
+    let currentLevel = theme;
+    while (propList.length > 0) {
+      const nextLevel = propList.shift();
+      if (nextLevel in currentLevel) {
+        if (typeof currentLevel[nextLevel] === 'string') {
+          return currentLevel[nextLevel] as string;
+        }
+        currentLevel = currentLevel[nextLevel] as Record<string, unknown>;
+      } else {
+        break;
+      }
+    }
+    return null;
```

Comment:
How does styled elements handle a null property value? Is that something we should catch with the compiler? Previously a typo or accessing a property in the wrong order (e.g. `theme.background.color.light` instead of `theme.color.background.light`) would be the caught by the compiler.

Reply:
Ah. That's a really good point. I believe that `styled-components` will just remove the css property if the value passed to it is null, so it would be like it was never defined. I don't know if there is a way we can catch those instances within Typescript itself, so we would have to give up the autocompletion features to use `withTheme`. I'll have to think on that a bit...

Reply:
My latest revisions in 6af3f48 adds the checking of subsequent values to `fromTheme`